ert_shared/libres_facade.py

```diff
@@ -81,3 +83,114 @@ def select_or_create_new_case(self, case_name):
         if self.get_current_case_name() != case_name:
             fs = self._enkf_main.getEnkfFsManager().getFileSystem(case_name)
             self._enkf_main.getEnkfFsManager().switchFileSystem(fs)
+
+    def cases(self):
+        return self._enkf_main.getEnkfFsManager().getCaseList()
+
+    def is_case_hidden(self, case):
+        return self._enkf_main.getEnkfFsManager().isCaseHidden(case)
+
+    def case_has_data(self, case):
+        return self._enkf_main.getEnkfFsManager().caseHasData(case)
+
+    def is_case_running(self, case):
+        return self._enkf_main.getEnkfFsManager().isCaseRunning(case)
+
+    def all_data_type_keys(self):
+        return self._enkf_main.getKeyManager().allDataTypeKeys()
+
+    def observation_keys(self, key):
+        if self._enkf_main.getKeyManager().isGenDataKey(key):
+            key_parts = key.split("@")
+            key = key_parts[0]
+            if len(key_parts) > 1:
+                report_step = int(key_parts[1])
+            else:
+                report_step = 0
+
+            obs_key = GenDataObservationCollector.getObservationKeyForDataKey(self._enkf_main, key, report_step)
+            if obs_key is not None:
+                return [obs_key]
+            else:
+                return []
+        elif self._enkf_main.getKeyManager().isSummaryKey(key):
+            return SummaryObservationCollector.observationKeys(self._enkf_main, key)
+        else:
+            return []
+
+    def gather_gen_kw_data(self, case, key):
+        """ :rtype: pandas.DataFrame """
+        data = GenKwCollector.loadAllGenKwData(self._enkf_main, case, [key])
+        return data[key].dropna()
+
+    def gather_summary_data(self, case, key):
+        """ :rtype: pandas.DataFrame """
+        data = SummaryCollector.loadAllSummaryData(self._enkf_main, case, [key])
+        if not data.empty:
+            data = data.reset_index()
+
+            if any(data.duplicated()):
+                print("** Warning: The simulation data contains duplicate "
+                      "timestamps. A possible explanation is that your "
+                      "simulation timestep is less than a second.")
+                data = data.drop_duplicates()
+
+            data = data.pivot(index="Date", columns="Realization", values=key)
+
+        return data #.dropna()
+
+    def has_refcase(self, key):
+        refcase = self._enkf_main.eclConfig().getRefcase()
+        return refcase is not None and key in refcase
+
+    def refcase_data(self, key):
+        refcase = self._enkf_main.eclConfig().getRefcase()
+
+        if refcase is None or key not in refcase:
+            return DataFrame()
+
+        values = refcase.numpy_vector(key, report_only=False)
+        dates = refcase.numpy_dates
+
+        data = DataFrame(zip(dates, values), columns=['Date', key])
+        data.set_index("Date", inplace=True)
+
+        return data.iloc[1:]
```

Comment:
There is quite extensive logic here which should be tested.

Reply:
Added test_libres_facade with tests for the new methods